src/overlays/actors/ovl_En_Floormas/z_en_floormas.c

```diff
@@ -91,144 +141,1105 @@ static DamageTable D_808D38D4 = {
+    }
+
+    this->unk_18E = Rand_S16Offset(2, 4);
+    this->actor.speedXZ = 1.5f;
+    this->actionFunc = func_808D0D70;
+}
+
+void func_808D0D70(EnFloormas* this, GlobalContext* globalCtx) {
+    s32 temp_v0;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    temp_v0 = func_801378B8(&this->skelAnime, 0.0f);
+    if ((temp_v0 != 0) && (this->unk_18E != 0)) {
+        this->unk_18E--;
+    }
+
+    if ((temp_v0 != 0) || func_801378B8(&this->skelAnime, 12.0f) || func_801378B8(&this->skelAnime, 24.0f) ||
+        func_801378B8(&this->skelAnime, 36.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_FALL_WALK);
+    }
+
+    if ((this->actor.xzDistToPlayer < 320.0f) && Actor_IsActorFacingLink(&this->actor, 0x4000)) {
+        func_808D0F50(this);
+    } else if (this->actor.bgCheckFlags & 8) {
+        this->unk_190 = this->actor.wallYaw;
+        func_808D108C(this);
+    } else if ((this->actor.xzDistToPlayer < 400.0f) && !Actor_IsActorFacingLink(&this->actor, 0x4000)) {
+        this->unk_190 = this->actor.yawTowardsPlayer;
+        func_808D108C(this);
+    } else if (this->unk_18E == 0) {
+        func_808D0ECC(this);
+    }
+}
+
+void func_808D0ECC(EnFloormas* this) {
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06009244);
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_808D0F14;
+}
+
+void func_808D0F14(EnFloormas* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_808D0B08(this);
+    }
+}
+
+void func_808D0F50(EnFloormas* this) {
+    this->unk_18E = 0;
+    this->actor.speedXZ = 5.0f;
+    this->skelAnime.animPlaybackSpeed = 3.0f;
+    this->actionFunc = func_808D0F80;
+}
+
+void func_808D0F80(EnFloormas* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_801378B8(&this->skelAnime, 0.0f) || func_801378B8(&this->skelAnime, 12.0f) ||
+        func_801378B8(&this->skelAnime, 24.0f) || func_801378B8(&this->skelAnime, 36.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_FALL_WALK);
+    }
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 3, 0x71C);
+    if ((this->actor.xzDistToPlayer < 280.0f) && Actor_IsActorFacingLink(&this->actor, 0x2000) &&
+        !(this->actor.bgCheckFlags & 8)) {
+        func_808D1380(this, globalCtx);
+    } else if (this->actor.xzDistToPlayer > 400.0f) {
+        func_808D0CE4(this);
+    }
+}
+
+void func_808D108C(EnFloormas* this) {
+    s16 sp36 = this->unk_190 - this->actor.shape.rot.y;
+
+    this->actor.speedXZ = 0.0f;
+    if (sp36 > 0) {
+        SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06002158, -3.0f);
+    } else {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06002158, -1.0f, SkelAnime_GetFrameCount(&D_06002158.common), 0.0f, 2,
+                             -3.0f);
+    }
+
+    if (this->actor.scale.x > 0.009f) {
+        this->unk_190 = sp36 * 0.06666667f;
```

Comment:
```suggestion
        this->unk_190 = sp36 * (1.0f / 15.0f)
```
I think it was suppose to be this